File: Tree/src/com/company/BinaryTree.java
package com.company;

public class BinaryTree
{
    public int[] nodes;

    public BinaryTree(int[] nodes)
    {
        this.nodes = nodes;
    }

    public Node CreateTree(Node root)
    {
        return  CreateTreeHelper(root,0);
    }

    private Node CreateTreeHelper(Node root,int i)
    {
        if(i<nodes.length)
        {
            Node n = new Node(nodes[i]);
            root = n;
            root.left = CreateTreeHelper(root.left,2*i+1);
            root.right = CreateTreeHelper(root.right,2*i+2);
        }
        return root;
    }

    public void inOrder(Node root)
    {
        if (root != null) {
            inOrder(root.left);
            System.out.print(root.value + " ");
            inOrder(root.right);
        }
    }
}

File: C++/Tree/BinaryTree/BinaryTree.cpp
//
// Created by ShubhraSarker on 9/19/2020.
//

#include "BinaryTree.h"
#include <bits/stdc++.h>

BinaryTree::BinaryTree(int nodes[], int length)
{
    this->nodes = nodes;
    this->length = length;
}

Node *BinaryTree::CreateTree(Node *root, int i)
{
    if(i>=length)
    {
        return nullptr;
    }
    Node* node = new Node(nodes[i]);
    root = node;
    root->leftNode = CreateTree(root->leftNode, 2 * i + 1);
    root->rightNode = CreateTree(root->rightNode, 2 * i + 2);
    return root;
}

void BinaryTree::InOrder(Node *root)
{
    if(root!= nullptr)
    {
        InOrder(root->leftNode);
        std::cout<<root->value << std::endl;
        InOrder(root->rightNode);
    }
}
void BinaryTree::PreOrder(Node *root)
{
    if(root!= nullptr)
    {
        std::cout<<root->value << std::endl;
        InOrder(root->leftNode);
        InOrder(root->rightNode);
    }
}
void BinaryTree::PostOrder(Node *root)
{
    if(root!= nullptr)
    {
        InOrder(root->leftNode);
        InOrder(root->rightNode);
        std::cout<<root->value << std::endl;
    }
}